sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj

```diff
@@ -25,4 +25,8 @@
     <AdditionalFiles Include="..\SonarLint.xml" Link="Properties\SonarLint.xml" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="System.ComponentModel.Composition" />

```

Comment:
You mean why is it needed? Without it, the code won't compile. `IsShared` method uses `CreationPolicy.Shared` from that assembly.

Reply:
I know why it is needed but this adds a dependency to the analyzer and I am not sure whether this is a problem or not.

Reply:
The MEF assembly will almost certainly be loaded by something else when the analyser is run inside VS (i.e. it will be in memory already), but it almost certainly **won't** be loaded when running under MSBuild, so this will cause an exception unless we start shipping the composition assembly with the analyser.

I think it would be better to just test the fully qualified name of the attribute as a string. There's no good reason anyone would have defined their own attribute with exactly the same qualified name.

Reply:
The problem is not the type of the attribute, but the value passed in the constructor. Class symbol provides only an int, which we need to interpret as one of `CreationPolicy` enum values. It is just a single int so I'll hardcode it with a comment <insert horrified face here>.